sdk/storage/src/datalake/path_client.cpp

```diff
@@ -251,37 +267,61 @@ namespace Azure { namespace Storage { namespace DataLake {
         m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
   }
 
-  PathUpdateResponse PathClient::SetProperties(const SetPathPropertiesOptions& options) const
+  SetPropertiesResult PathClient::SetProperties(const SetPathPropertiesOptions& options) const
   {
     DataLakeRestClient::Path::UpdateOptions protocolLayerOptions;
     // TODO: Add null check here when Nullable<T> is supported
     protocolLayerOptions.Action = PathUpdateAction::SetProperties;
-    protocolLayerOptions.CacheControl = options.CacheControl;
-    protocolLayerOptions.ContentType = options.ContentType;
-    protocolLayerOptions.ContentDisposition = options.ContentDisposition;
-    protocolLayerOptions.ContentEncoding = options.ContentEncoding;
-    protocolLayerOptions.ContentLanguage = options.ContentLanguage;
+    protocolLayerOptions.CacheControl = options.HttpHeaders.CacheControl;
+    protocolLayerOptions.ContentType = options.HttpHeaders.ContentType;
+    protocolLayerOptions.ContentDisposition = options.HttpHeaders.ContentDisposition;
+    protocolLayerOptions.ContentEncoding = options.HttpHeaders.ContentEncoding;
+    protocolLayerOptions.ContentLanguage = options.HttpHeaders.ContentLanguage;
     protocolLayerOptions.IfMatch = options.IfMatch;
     protocolLayerOptions.IfNoneMatch = options.IfNoneMatch;
     protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
     protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
     protocolLayerOptions.Properties = Details::SerializeMetadata(options.Metadata);
     protocolLayerOptions.Timeout = options.Timeout;
-    return DataLakeRestClient::Path::Update(
-        m_dfsUri.ToString(), *m_pipeline, options.Context, nullptr, protocolLayerOptions);
+    auto emptyStream = Azure::Core::Http::NullBodyStream();
+    auto result = DataLakeRestClient::Path::Update(
+        m_dfsUri.ToString(), emptyStream, *m_pipeline, options.Context, protocolLayerOptions);
+    auto returnVal = SetPropertiesResult{};
+    returnVal.Date = std::move(result.Date);
+    returnVal.RequestId = std::move(result.RequestId);
+    returnVal.Version = std::move(result.Version);
+    returnVal.ETag = std::move(result.ETag);
+    returnVal.LastModified = std::move(result.LastModified);
+    returnVal.HttpHeaders = std::move(result.HttpHeaders);
+    returnVal.ContentLength = result.ContentLength;
+    returnVal.ContentRange = std::move(result.ContentRange);
+    returnVal.ContentMD5 = std::move(result.ContentMD5);
+    auto metadata = Details::DeserializeMetadata(result.Properties);
+    if (metadata.HasValue())
+    {
+      returnVal.Metadata = metadata.GetValue();
```

Comment:
Why not make metadata in protocol layer non-nullable?

Reply:
Did that.